File: Algorithms/Python/bubble_sort.py
from __future__ import annotations

from typing import TypeVar

T = TypeVar("T", bound=int | float)


def bubble_sort(arr: list[T]) -> list[T]:
    """
    Bubble sort algorithm to sort the array in ascending order.
    """
    for i in range(len(arr)):
        for j in range(len(arr) - 1, i, -1):
            if arr[j] < arr[j - 1]:
                arr[j], arr[j - 1] = arr[j - 1], arr[j]

    return arr


if __name__ == "__main__":
    arr = [2, 5, 3, 1, 4, 6]
    print(bubble_sort(arr))  # [1, 2, 3, 4, 5, 6]

File: Algorithms/Go/bubble_sort.go
package main

import "fmt"

func bubbleSort(arr []int) {
	for i := 0; i < len(arr); i++ {
		for j := 1; j < len(arr)-i; j++ {
			if arr[j-1] > arr[j] {
				arr[j-1], arr[j] = arr[j], arr[j-1]
			}
		}
	}
}

func main() {
	arr := []int{5, 4, 3, 2, 1}
	bubbleSort(arr)
	fmt.Println(arr)
}
